### DISCUSSION TITLE
Does the Mac support Uart baudrate change to 3M?

### DISCUSSION THREAD

**Discussion by yxq-Jason** [armadsen/ORSSerialPort#172] [Q&A] [answered]
Hi
I am trying to run a DOE with select baudrate change from 230400 to 3M in Mac.
it works well except for 3M baudrate.(coolTerm not work as well) 
<img width="527" alt="Screen Shot 2021-02-01 at 9 56 58 AM" src="https://user-images.githubusercontent.com/33271397/106406482-5d588b80-6474-11eb-9ee9-0c5f210ceb6f.png">
Something weird is it works in Win10 with coolTerm.
<img width="689" alt="Screen Shot 2021-01-29 at 2 58 05 PM" src="https://user-images.githubusercontent.com/33271397/106406548-8f69ed80-6474-11eb-9935-4f28050e90c3.png">

My question is does the Mac support 3M baudrate change? I don't know the difference between Mac and Windows. 
Is there something missing?

**Comment by yxq-Jason** [comment]
Add more info. Here I use FT4232H to test high speed with 3M baudrate. I use ORSSerial library or coolTerm to switch 3M baudrate but all not work in Mac(got nil response), only works in Windows. Has anyone tried?

**Comment by armadsen** [comment, marked as answer]
This is a function of the driver and specific USB to serial hardware you're using. You can try setting `serialPort.allowsNonStandardBaudRates = true` and then setting your high baud rate using the `baudRate` property, but that may or may not work, and the underlying system API that ORSSerialPort uses doesn't really provide any direct feedback on whether it did or not.
